tests/UtilTests/SignerTests.elm

```diff
@@ -0,0 +1,67 @@
+module UtilTests.SignerTests exposing (all)
+
+import AWS exposing (..)
+import AWS.Config
+import AWS.Http exposing (RequestParams(..), unsignedRequest)
+import AWS.Signers.V4 exposing (..)
+import Date exposing (fromString, fromTime)
+import Expect
+import Json.Decode as JD
+import Test exposing (describe, test, Test)
+import Time
+
+
+all : Test
+all =
+    describe "Signer"
+        [ authorizationTests
+        ]
+
+
+authorizationTests : Test
+authorizationTests =
+    describe "authorization"
+        {- http://docs.aws.amazon.com/general/latest/gr/signature-v4-test-suite.html#signature-v4-test-suite-example -}
+        [ test "returns a signed header equal to the online example" <|
+            \_ ->
+                let
+                    creds =
+                        AWS.Config.Credentials "AKIDEXAMPLE" "wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY" Nothing
+
+                    date =
+                        case (fromString "Sun, 30 Aug 2015 12:36:00") of
+                            Ok date ->
+                                date
+
+                            Err _ ->
+                                fromTime Time.hour
```

Comment:
This is a bit of a lazy hack. Now that `fromString` has been validated once for this test, it should never fail, but if it does, you'll see an incorrect date show up in the test output from the failed `Expect` call